lib/integrations/slack/incoming_message_builder.rb

```diff
@@ -17,6 +18,8 @@ def perform
       verify_hook
     elsif create_message?
       create_message
+    elsif link_shared?
+      Integrations::Slack::SlackLinkUnfurlService.new(params: params, integration_hook: integration_hook).perform
```

Comment:
A suggestion here would be to make it a job rather than running it in synchronously in the webhook callback itself.